techniques/systemSettings/networking/firewall/1.0/firewall.st

```diff
@@ -0,0 +1,99 @@
+# Here we have our firewall configuration (iptables or nftables)
+# But format may not be exactly what the output of iptables-save or nft export.
+# What we will do it to maintain an export from a previous run to be able to compare
+# with current export and detect changes.
+# This way we will be able to report config diffs and fix the config when necessary.
+
+bundle agent firewall_&RudderUniqueID& {
+  vars:
+      "config"    string => "&FIREWALL_CONF&";
+      "firewall"  string => "rudder_&FIREWALL_IMPL&";
+      "posthook"  string => "&FIREWALL_POST_HOOK_COMMAND&";
+      "reference" string => "/var/rudder/tmp/firewall.ref";
+
+  methods:
+      "any" usebundle => rudder_set_force_component_none("true");
+      "en1" usebundle => enable_reporting;
+      # Package
+      "any" usebundle => _method_reporting_context("Installation", "None");
+      "any" usebundle => package_present("${${firewall}.package}", "", "", "");
+      # Service
+      "any" usebundle => _method_reporting_context("Service activation", "None");
+      "any" usebundle => service_enabled("${${firewall}.service}");
+      "any" usebundle => _method_reporting_context("Service start", "None");
```

Comment:
what happens if service was stopped, and we start it but our template is broken. We end up with a service up and unconfigured, which could be problematic

Reply:
Indeed, it should be started after config change